### PR TITLE
feat: rock paper scissors refactor

### PR DESCRIPTION
**Pull request by ananas-block**
Versions:
- bump cli to alpha 18
- bump zk.js to alpha 13

Feat:
- enable PSPs to use pdas
- commented blinding check because it a usability security check which prevents our hardcoded blinding in rock paper scissors
- added checkAppData to throw an error if an appData attribute is not set
- support for multiple appUtxos
- support for additonal native in utxos
- auto generation of isAppInUtxo array

Fix:
- getVerifierProgramId was broken because anchor includes all constants in alphabetical order in the idl now
- createOutUtxos - 
- compileAndProve transactions enable controlling shuffling of utxos - (shuffling breaks psp tx right now)

### CODE REVIEW

File: light-zk.js/src/transaction/transaction.ts [hunk 1]
```diff
@@ -820,12 +820,18 @@ export class Transaction {
       );
 
       let inputsVec = (await coder.encode(accountName, inputs)).subarray(8);
+      // TODO: check whether app account names overlap with system account names and throw an error if so
+      let appAccounts = {};
```
Comment (SwenSchaeferjohann) [inline light-zk.js/src/transaction/transaction.ts:824]: shouldnt this be an array?

File: light-zk.js/src/types/programParameters.ts [hunk 2]
```diff
@@ -4,4 +4,5 @@ export type ProgramParameters = {
   verifierIdl: Idl;
   inputs: any; // object of proof and other inputs
   path: string;
+  accounts?: any; // object of accounts
```
Comment (SwenSchaeferjohann) [inline light-zk.js/src/types/programParameters.ts:7]: why not replace `any` with the type?

File: light-zk.js/src/utxo.ts [hunk 3]
```diff
@@ -132,16 +132,16 @@ export class Utxo {
...
+    //   throw new UtxoError(
+    //     UtxoErrorCode.BLINDING_EXCEEDS_SIZE,
+    //     "constructor",
 
-        `Bliding ${blinding}, exceeds size of 31 bytes/248 bit.`,
-      );
-    }
+    //     `Bliding ${blinding}, exceeds size of 31 bytes/248 bit.`,
+    //   );
+    // }
```
Comment (SwenSchaeferjohann) [inline light-zk.js/src/utxo.ts:144]: can this be removed?

**Review by SwenSchaeferjohann — COMMENTED** [review]
lgtm once above questions resolved  can merge

File: light-zk.js/src/transaction/transaction.ts (hunk 1 above)
Comment (ananas-block) [reply light-zk.js/src/transaction/transaction.ts:824]: Anchor requires an account object. This way we can just do:
```
tx.accounts({...appAccounts})
```

File: light-zk.js/src/types/programParameters.ts (hunk 2 above)
Comment (ananas-block) [reply light-zk.js/src/types/programParameters.ts:7]: We don't know what accounts are added in a psp before hand. The type needs to be generic. In the future we should make it depend on the idl.

File: light-zk.js/src/utxo.ts (hunk 3 above)
Comment (ananas-block) [reply light-zk.js/src/utxo.ts]: Good point I replaced the check with a better one

File: light-zk.js/src/utxo.ts [hunk 4]
```diff
@@ -131,17 +132,13 @@ export class Utxo {
...
-    //     "constructor",
-
-    //     `Bliding ${blinding}, exceeds size of 31 bytes/248 bit.`,
-    //   );
-    // }
+    if (!blinding.eq(blinding.mod(FIELD_SIZE))) {
+      throw new UtxoError(
+        UtxoErrorCode.BLINDING_EXCEEDS_FIELD_SIZE,
+        "constructor",
+        `Bliding ${blinding}, exceeds field size.`,
```
Comment (SwenSchaeferjohann) [inline light-zk.js/src/utxo.ts:139]: just small typo bliding

Comment (ananas-block) [reply light-zk.js/src/utxo.ts:139]:
```suggestion
        `Blinding ${blinding}, exceeds field size.`,
```

**Review by SwenSchaeferjohann — APPROVED** [review]

**Pull request merged**